keps/sig-api-machinery/2340-Consistent-reads-from-cache/README.md

```diff
@@ -207,6 +216,13 @@ Since falling back to etcd won't work, we should fail the requests and rely on
 rate limiting to prevent cascading failure.  I.e. `Retry-After` HTTP header (for
 well behaved clients) and [Priority and Fairness](https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20190228-priority-and-fairness.md).
 
+For such situations we will provide a dedicated metric to detect a problem,
+and a per-request override for debugging. If there is a ongoing problem,
+user can reconfigure kube-apiserver to always pass request to etcd by 
+disabling the feature flag. Metric `apiserver_watch_cache_read_wait`
```

Comment:
Once it graduates to GA, it's not possible anymore (the FG are LockedToDefault).

Does it mean we should expose an option an explicit flag to disable it?

Reply:
Not if we block GA on dropping support of supporting etcd v3.3.

Reply:
Well... maybe, but I'm not fully convinced here.

Reply:
Or we could introduce a flag, wait for etcd to release v3.8 and deprecate it then.

Reply:
So we basically need an explicit flag - I would make this explicit (and probably add that explicitly to graduation criteria for GA).

Reply:
Do we need a feature flag then? Both will flip the same feature.

Reply:
> Do we need a feature flag then? Both will flip the same feature.

FeatureGates force consumers to acknowledge the stability level of the flag they are specifying and provide an easy way to express things like, "only use stable features".  Flags don't allow that, so we require both for server features.

Reply:
Ok, makes sense. I etcd we use "experimental" flag prefix, but combining flag name with maturity is bad.